ironic_prometheus_exporter/messaging.py

```diff
@@ -0,0 +1,30 @@
+import os
+import json
+
+from oslo_config import cfg
+from oslo_messaging.notify import notifier
+
+prometheus_opts = [
+    cfg.StrOpt('file_path', required=True,
+               help='Path for the json file where the metrics will be stored.')
+]
+
+
+def register_opts(conf):
+    conf.register_opts(prometheus_opts, group='oslo_messaging_notifications')
+
+
+class PrometheusFileDriver(notifier.Driver):
+    """Publish notifications into a File to be used by Prometheus"""
+
+    def __init__(self, conf, topics, transport):
+        self.file_path = conf.oslo_messaging_notifications.file_path
+        if not self.file_path.endswith('.json'):
+            raise Exception('The file should end with .json')
+        if not os.path.exists(os.path.dirname(self.file_path)):
+            os.makedirs(os.path.dirname(self.file_path))
+        super(PrometheusFileDriver, self).__init__(conf, topics, transport)
+
+    def notify(self, ctxt, message, priority, retry):
```

Comment:
Would it make sense to loop over (potentially) failing write respecting the `retry` parameter?

Reply:
Hi Ilya, the notify method is an abstract method defined in the Driver class in oslo.messaging [1]
so we keep the same parameters 

[1] https://github.com/openstack/oslo.messaging/blob/master/oslo_messaging/notify/notifier.py#L125

Reply:
o/
Right, we are implementing this method, it has the `retry` parameter:

```
        :param retry: connection retries configuration (used by the messaging
                      driver):
                      None or -1 means to retry forever.
                      0 means no retry is attempted.
                      N means attempt at most N retries.
```

It seems we currently ignore `retry`, so I wonder if we should respect it in one way or the other...?

Reply:
This is the initial version for the driver, we just want the messages written to the json file, I think it's  not necessary for now. For example the log driver doesn't use

https://github.com/openstack/oslo.messaging/blob/648ad56c8652d83d3b1cae745488bb8e0450febd/oslo_messaging/notify/_impl_log.py

Reply:
I don't think we expect writing to local files to fail in a retryable way :)

Reply:
Well, "No space left on device" seems theoretically possible and potentially recoverable..

I am just thinking how could we behave in a way the abstract interface expects us to...

Reply:
Yes, but retrying won't help until somebody cleans up the disk :)

Reply:
Ironic lands are inhabited by daemons. One might clean things up, eventually. Also, the work flow can consume file system space temporarily.

Just thinking aloud, that's nit indeed.